File: labs/step-functions/aws/python/src/validate/handler.py
import re


def lambda_handler(event, context):
    raw = str(event.get("cep", "")).strip()
    digits = re.sub(r"\D", "", raw)

    if len(digits) != 8:
        raise ValueError(f"CEP inválido: '{raw}' — esperado 8 dígitos")

    return {
        "cep": digits,
        "normalized": True,
        "simulateFailure": bool(event.get("simulateFailure", False)),
    }

File: labs/step-functions/aws/nodejs/src/validate/handler.js
exports.handler = async (event) => {
  const raw = String(event.cep ?? "").trim();
  const digits = raw.replace(/\D/g, "");

  if (digits.length !== 8) {
    throw new Error(`CEP inválido: '${raw}' — esperado 8 dígitos`);
  }

  return {
    cep: digits,
    normalized: true,
    simulateFailure: Boolean(event.simulateFailure),
  };
};
